### PR TITLE
feature: implement the back end for play flashcard

### LINKED ISSUE #209: FlashCard : Add a sub collection in user to store information about the user knowledge of his flashcards

### PR DESCRIPTION
**Pull request by LawrenceLeitgib**
# Description

This PR implemente the backend to play the flashcard with a data class represting the user informations for each flashcard (e.g. last reviewed and level) It add methodes to the user view model and to the user firestore repository to simplying interaction from the UI to to the back end

Fixes #209

# How Has This Been Tested?
Unit  have been added to test the new methodes added to the user view model and repository

### Unit Tests
Tests were added in `UserRepositoryFirestoreTest.kt` and `UserViewModelTest.kt` to test each new methode added
  
# Dependencies
No dependecies

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged in downstream modules, no conflicts with main

### CODE REVIEW

**Review by robertaleG — CHANGES_REQUESTED** [review]
The PR is missing some docs and needs some minor tweaks.  The User MVVM  tests are also a bit all over the place (unnecessary assertions and missing necessary ones).  Apart from that you did a good job 👍

File: app/src/test/java/com/github/onlynotesswent/model/user/UserViewModelTest.kt [hunk 1]
```diff
@@ -550,4 +580,75 @@ class UserViewModelTest {
...
+            lastModified = Timestamp.now(),
+            flashcardIds = listOf("flashcardId"))
+
+    val flashcard = UserFlashcard("flashcardId", 0, Timestamp.now())
+    userViewModel.addUser(user)
+
+    userViewModel.updateUserFlashcard(flashcard, { assert(true) }, { assert(false) })
+
+    // verifies that the default parameters work
+    userViewModel.updateUserFlashcard(flashcard)
```
[the remark this answers is not in the document]
Comment (robertaleG) [reply app/src/test/java/com/github/onlynotesswent/model/user/UserViewModelTest.kt:633]: forgot to put int the review but you did the same again for the delete function, I think it shouldn't be necessary and will just slow down the CI. a default value {} for a callback parameter can't really fail a test

File: app/src/test/java/com/github/onlynotesswent/model/user/UserViewModelTest.kt [hunk 2]
```diff
@@ -550,4 +580,75 @@ class UserViewModelTest {
...
+
+    // adding get the deck to test the update
+    userViewModel.getUserFlashcardFromDeck(deck, { assert(false) })
```
[the remark this answers is not in the document]
Comment (LawrenceLeitgib) [reply app/src/test/java/com/github/onlynotesswent/model/user/UserViewModelTest.kt:636]: I am testing the if statement inside the update methode

**Review by robertaleG — APPROVED** [review]
Very nice changes, well done. I will merge

**Pull request merged**